pkgs/top-level/python-packages.nix

```diff
@@ -80,10 +84,21 @@ let
         else throw "Unsupported kind ${kind}");
     in fetcher (builtins.removeAttrs attrs ["format"]) );
 
+  # Check whether a derivation provides a Python module.
+  hasPythonModule = drv: (hasAttr "pythonModule" drv) && ( (getAttr "pythonModule" drv) == python);
+
+  # Recurse into a list of Python modules, returning all Python modules that are required.
+  requiredModules = drvs: filter hasPythonModule (closePropagation drvs);
```

Comment:
Maybe name it ```requiredPythonModules``` or even ```filterPythonModules```. The meaning of ```requiredModules``` is hard to understand if you don't think in the context of Python modules.